torch/_guards.py

```diff
@@ -53,16 +53,24 @@
 
 
 class CompileId(NamedTuple):
-    frame_id: int
+    compiled_autograd_id: Optional[int]
+    frame_id: Optional[int]
     # This id is per-frame, and counts how many times we've compiled this
     # frame.  This could have been a global id but having this be per-frame
     # gives you a better intuitive sense for how many recompiles have occurred
     # so far.
-    frame_compile_id: int
+    frame_compile_id: Optional[int]
     # TODO: consider also tracking the recompilation count
 
     def __str__(self):
-        return f"{self.frame_id}/{self.frame_compile_id}"
+        if self.compiled_autograd_id is not None:
+            assert (self.frame_id is None) == (self.frame_compile_id is None)
+            return f"{self.compiled_autograd_id}/{self.frame_id}/{self.frame_compile_id}".replace(
+                "None", "-"
+            )
+        else:
+            assert self.frame_id is not None and self.frame_compile_id is not None
+            return f"{self.frame_id}/{self.frame_compile_id}"
```

Comment:
See also comments from tlparse. The format here is not so easy to change so let's make sure we bikeshed it enough.

Reply:
I am going with adding a prefix symbol "!" for compiled autograd